adafruit_displayio_layout/widgets/icon_widget.py

```diff
@@ -52,19 +66,47 @@ class IconWidget(Widget, Control):
     :param int max_size: (Optional) this will get passed through to the
      displayio.Group constructor. If omitted we default to
      grid_size width * grid_size height to make room for all (1, 1) sized cells.
-
     """
 
-    def __init__(self, label_text, icon, on_disk=False, **kwargs):
-        super().__init__(**kwargs)
+    # pylint: disable=bad-super-call, too-many-instance-attributes, too-many-locals
+    # pylint: disable=too-many-arguments, unused-argument
+
+    _max_scale = 1.5
+    _max_pixels = 80
+    _max_color_depth = 512
+    _zoom_bitmap = Bitmap(
+        round(_max_scale * _max_pixels),
+        round(_max_scale * _max_pixels),
+        _max_color_depth,
+    )
+    _zoom_palette = Palette(_max_color_depth)
+
+    def __init__(
+        self,
+        display,
```

Comment:
Can we make `display` a keyword arg and add checks where it is used for None. I'm curious if the animation would still work if the user had auto_show enabled on the display object. If so it would be nice to allow users to skip this argument if they want to.

Reply:
Ok, I updated the structure so that the display is defined in a Class initialization method.